moveit_planners/ompl/ompl_interface/src/planning_context_manager.cpp

```diff
@@ -473,11 +473,20 @@ ompl_interface::ModelBasedPlanningContextPtr ompl_interface::PlanningContextMana
   // However consecutive IK solutions are not checked for proximity at the moment and sometimes happen to be flipped,
   // leading to invalid trajectories. This workaround lets the user prevent this problem by forcing rejection sampling
   // in JointModelStateSpace.
+  //
+  // Additionally, check if the requested planner is of the informed planner family (AITstar, ABITstar, BITstar) that
+  // does not support PoseModelStateSpace. If yes, force planning with JointModelStateSpace.
   ModelBasedStateSpaceFactoryPtr factory;
   auto it = pc->second.config.find("enforce_joint_model_state_space");
 
   if (it != pc->second.config.end() && boost::lexical_cast<bool>(it->second))
     factory = getStateSpaceFactory(JointModelStateSpace::PARAMETERIZATION_TYPE);
+  else if (req.planner_id.find("AITstar") != std::string::npos)
+    factory = getStateSpaceFactory(JointModelStateSpace::PARAMETERIZATION_TYPE);
+  else if (req.planner_id.find("ABITstar") != std::string::npos)
+    factory = getStateSpaceFactory(JointModelStateSpace::PARAMETERIZATION_TYPE);
+  else if (req.planner_id.find("BITstar") != std::string::npos)
+    factory = getStateSpaceFactory(JointModelStateSpace::PARAMETERIZATION_TYPE);
   else
     factory = getStateSpaceFactory(pc->second.group, req);
```

Comment:
I added the following checks to enforce JointModelStateSpace when using any of the three Informed planners.

I'm a pretty poor C++ coder and don't know much about the inner workings of MoveIt, so I would be happy if someone more proficient could double check. :pray:

Reply:
I believe the planner_id can be any name the user wants to use in the `ompl_planning.yaml` file (even though it is usually the planner name, sometimes one can have two configurations for the same plan and use custom planner_ids to differentiate them). I reckon the `type` parameter is the one you are looking for.

```yaml
AITstar:    <-- This is the planner_id, which the user can define as they want
    samples_per_batch: 1000
    use_graph_pruning: true
    use_k_nearest: true
    rewire_factor: 1.001
    find_approximate_solutions: false
    set_max_num_goals: 10
    type: geometric::AITstar  <-- This is the planner type
```

You probably would need to do something like
```cpp
  auto type_it = pc->second.config.find("type");
  std::string planner_type;
  if (type_it != pc->second.config.end())
    planner_type = type_it->second;
```
to get the planner type, and then make the following comparison:
```cpp
else if (planner_type == "geometric::AITstar")
```

Finally, you probably will need to check with the maintainers if they agree with this change, or if they have a better suggestion.

Reply:
Good point! I modified accordingly.
Thanks for the suggestion.